### PR TITLE
throw errors when we get them

### PR DESCRIPTION
**Pull request by jonstorer**

### CODE REVIEW

File: spec/integration/account_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          :status => 200,
+          :body => fixture('account'),
+          :headers => { 'Content-Type': 'application/json' }
+        })
+    end
+
+    it 'makes the correct call to update' do
+      account = Cashbox::Account.new({ id: 1 })
+      account.save
+      account.save
```
Comment (noheltcj) [inline spec/integration/account_spec.rb]: extra call?

File: lib/cashbox/repository.rb [hunk 2]
```diff
@@ -67,6 +67,8 @@ def cast(hash)
         hash['data'].map do |data|
           @instance.class.new(data)
         end
+      when 'Error'
```
Comment (maulin) [inline lib/cashbox/repository.rb]: could also
```ruby
def cast
  //do stuff
  rescue => e
    raise Cashbox::Error.new(hash)
end
```

File: spec/integration/account_spec.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        .to_return({
+          :status => 200,
+          :body => fixture('account'),
+          :headers => { 'Content-Type': 'application/json' }
+        })
+
+      account = JSON.parse(fixture('account'))
+      account['payment_methods'] = account['payment_methods']['data']
+
+      stub_post("/accounts/#{account['vid']}")
```
Comment (noheltcj) [inline spec/integration/account_spec.rb]: is this both a stub and an expectation?

File: lib/cashbox/repository.rb (hunk 2 above)
Comment (jonstorer) [reply lib/cashbox/repository.rb]: I'll do both. However there is a known object named "Error" and we should solve for it.

File: spec/integration/account_spec.rb (hunk 3 above)
Comment (jonstorer) [reply spec/integration/account_spec.rb]: kinda both,,, it's a stub that returns an object that can be asserted against

File: lib/cashbox/repository.rb (hunk 2 above)
Comment (jonstorer) [reply lib/cashbox/repository.rb]: I'm going to hold off on that, I want to have a better plan for the exception was throw approach

**Pull request closed without merging**